semantic-python/src/Language/Python/Core.hs

```diff
@@ -161,7 +159,7 @@ instance Compile Py.Assignment where
     , ann
     } cc next = do
     (names, val) <- desugar [Located ann name] rhs
-    compile val pure next >>= foldr collapseDesugared cc names >>= locate it
+    locate it <$> compile val pure next >>= foldr collapseDesugared cc names
```

Comment:
Note that this changes the semantics due to the relative precedence of `<$>` and `>>=`; you probably want to parenthesize the RHS or else use `>>= pure . locate it`.